taichi/program/program.cpp

```diff
@@ -256,6 +256,19 @@ void Program::materialize_layout() {
   }
 }
 
+void Program::check_runtime_error() {
+  TI_ASSERT(arch_is_cpu(config.arch));
+  auto tlctx = llvm_context_host.get();
+  auto runtime_jit_module = tlctx->runtime_jit_module;
+  runtime_jit_module->call<void *>("retrieve_error_code", llvm_runtime);
+  auto error_code = runtime_jit_module->fetch_result<int64>();
+  if (error_code) {
+    runtime_jit_module->call<void *>("retrieve_error_message", llvm_runtime);
+    auto error_message = runtime_jit_module->fetch_result<char *>();
+    TI_ERROR("Assertion failure: {}", error_message);
```

Comment:
Still using `TI_ERROR` for now. How to make it be captured by python?

Reply:
In the test case you can set `ti.set_gdb_trigger(False)` and then Taichi will simply raise a Python exception instead of triggering gdb.